File: twitterClient/twitterClient/src/stomp/frames/ConnectedFrame.cpp
//
//  ConnectedFrame.cpp
//  Object to represent the CONNECTED frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/ConnectedFrame.h"

namespace stomp {
    ConnectedFrame::ConnectedFrame(): StompFrame(CONNECTED_FRAME_NAME) { }
    
    ConnectedFrame::ConnectedFrame(std::string version): StompFrame(CONNECTED_FRAME_NAME) {
        addHeaderPair("version", version);
    }
    
    void ConnectedFrame::accept(FrameVisitor *visitor)
    {
        visitor->visit(this);
    }
    
}

File: twitterEngine/src/spl/stomp/protocol/ConnectedFrame.java
package spl.stomp.protocol;

/**
 * @name Connected Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * CONNECTED stomp frame
 */
class ConnectedFrame extends StompFrame {
    public ConnectedFrame(String session) {
        super(StompCommand.CONNECTED);
        addFrameHeader("session", session);
        addFrameHeader("version", "1.2");
    }
}
